lib/meta-box-partial-page.php

```diff
@@ -324,10 +324,19 @@ function the_gutenberg_metaboxes() {
 	 * editor instance. If a cleaner solution can be imagined, please change
 	 * this, and try to get this data to load directly into the editor settings.
 	 */
-	wp_add_inline_script(
-		'wp-edit-post',
-		'window._wpLoadGutenbergEditor.then( function( editor ) { editor.initializeMetaBoxes( ' . wp_json_encode( $meta_box_data ) . ' ) } );'
-	);
+	$script = 'window._wpLoadGutenbergEditor.then( function( editor ) { editor.initializeMetaBoxes( ' . wp_json_encode( $meta_box_data ) . ' ) } );';
+
+	wp_add_inline_script( 'wp-edit-post', $script );
+
+	/**
+	 * When `wp-edit-post` is output in the `<head>`, the inline script needs to be manually printed. Otherwise,
```

Comment:
Can't we force `wp-edit-post` to print in the footer? I thought the last `true` argument serves this purpose?

Reply:
`wp-edit-post` will default to printing in the footer, but if a script that prints in the header depends on it, then `WP_Script`'s dependency handling will move it to the header.

Reply:
Got it, makes me wonder if there should be a way to "force" dependencies in the footer as well. Thanks for the clarification.

Reply:
It's one of those back compat issues. Historically, JS would be written in a way that required it to be loaded in the header. Then more modern practices came along, JS would be written to run asynchronously with the page render, so it could be loaded in the footer, instead.

However, we still had a world of JS that needed to be run synchronously. And if that JS required some other JS that could run asynchronously, the only safe option was to move the async JS to the header.

Reply:
@youknowriad Check out https://core.trac.wordpress.org/ticket/44709. Hoping to find a better alternative long term solution to this issue.